tensorflow-ops/src/TensorFlow/Variable.hs

```diff
@@ -30,13 +30,18 @@ import TensorFlow.Core
 import TensorFlow.Build (opDef)
 import TensorFlow.BuildOp (buildInputs, pureOp, OpParams)
 import TensorFlow.Output (opInputs, unNodeName)
-import TensorFlow.Tensor (tensorNodeName)
+import TensorFlow.Tensor (tensorNodeName, renderedOutput)
 import TensorFlow.Types (tensorType)
 import qualified TensorFlow.GenOps.Core as CoreOps
-import TensorFlow.Ops (zeros)
+import TensorFlow.Ops (zeros, fill, shape)
+import TensorFlow.Gradient (GradientTarget(..))
 
 newtype Variable a = Variable (Tensor Value ResourceHandle)
 
+instance GradientTarget Variable where
+    targetOutput (Variable v) = renderedOutput v
+    targetZeros (Variable v) = fill (shape v) 0
```

Comment:
I *think* this won't do what's intended.  A Variable is a multidimensional array of `DT_RESOURCE` type whose dimensions happen to contain only one element, so as far as I can tell this will always return a constant size.

See for example the implementation of ReadHandle:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/resource_variable_ops.cc#L44
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/resource_mgr.cc#L246

I think we want `zerosLike (readValue v)` here instead?  Which might suggest that the class method we want is closer to (`t a -> Tensor Value a`)?

(Even better if you can add a test that confirms whether my understand is correct.)